nomad/fsm.go

```diff
@@ -204,6 +207,49 @@ func (n *nomadFSM) applyUpsertJob(buf []byte, index uint64) interface{} {
 		n.logger.Printf("[ERR] nomad.fsm: UpsertJob failed: %v", err)
 		return err
 	}
+
+	// If it is periodic, insert it into the periodic runner and record the
+	// time it was inserted.
+	if req.Job.IsPeriodic() {
```

Comment:
If the job is not periodic but was previously, we need to remove it from the periodic runner